web-wallet/src/lib/dusk/string/validateAddress.js

```diff
@@ -0,0 +1,22 @@
+/**
+ * Validates a Dusk address, with feedback on failure or success.
+ * This function is carried over from
+ * [dusk-wallet-js repo](https://github.com/dusk-network/dusk-wallet-js/blob/71ac65a5105757768bf1af56898a7969a3e06c11/src/keys.js)
+ * @param {String} address The public spent key to validate.
+ * @returns {{isValid: boolean, reason: string}} An object with two keys:
+ *  - `isValid` {Boolean} - true if the address is valid, false if invalid.
+ *  - `reason` {String} - describes why the address is invalid or confirms if it is valid.
+ */
+export default function validateAddress (address) {
+	const regex = /^[123456789ABCDEFGHJKLMNPQRSTUVWXYZabcdefghijkmnopqrstuvwxyz]{87,88}$/;
```

Comment:
Some questions here @HDauven :
- the capital "i" and the lower case "l" are excluded on purpose because of visual similarity, right? (I would use character ranges anyway to help readability)
- why do you make a regexp to test the whole address (length included) if the length is checked otherwise to give a different error message? Isn't it faster to [search](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/search) for the first occurrence of an invalid character (haven't tried, and performance won't matter much here, but still)? 

But more importantly: why don't we make a PR to have the `dusk-wallet-js` expose this function instead of maintaining a copy of it here?

Reply:
> But more importantly: why don't we make a PR to have the `dusk-wallet-js` expose this function instead of maintaining a copy of it here?

Matteo has responded [here](https://github.com/dusk-network/dusk-wallet-js/pull/56#issuecomment-1954015211).

Based on that, Hein and I discussed the best way going forward, at least for the time being, might be porting the validator as a utility function in our code – as we don't know what the plan is, if any, to have it accessible from elsewhere, while I think that allowing the user to progress with a malformed address is bad experience that we could easily resolve.

Reply:
> > But more importantly: why don't we make a PR to have the `dusk-wallet-js` expose this function instead of maintaining a copy of it here?
> 
> Matteo has responded [here](https://github.com/dusk-network/dusk-wallet-js/pull/56#issuecomment-1954015211).
> 
> Based on that, Hein and I discussed the best way going forward, at least for the time being, might be porting the validator as a utility function in our code – as we don't know what the plan is, if any, to have it accessible from elsewhere, while I think that allowing the user to progress with a malformed address is bad experience that we could easily resolve.

Ok, my bad: I thought it was merged there.

Reply:
> Some questions here @HDauven :
> 
> * the capital "i" and the lower case "l" are excluded on purpose because of visual similarity, right? (I would use character ranges anyway to help readability)

Correct, same with o/O/0.

> * why do you make a regexp to test the whole address (length included) if the length is checked otherwise to give a different error message? Isn't it faster to [search](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/search) for the first occurrence of an invalid character (haven't tried, and performance won't matter much here, but still)?

You sure can do that. I ported it from a different use case and morphed it into a multi-check. Checking for invalid characters would be a great improvement in my opinion, since it'll explicitly show which characters are invalid towards the reader of the code.

Reply:
I have tweaked the regex check with the `search` method. Have a look [here](https://github.com/dusk-network/rusk/blob/3258ad8494974fb111b7323cbe096d410dbb3d80/web-wallet/src/lib/dusk/string/validateAddress.js).

cc: @ascartabelli